File: cases/taint/go/exception-catch-negative/exception_catch_negative.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: exception-catch-input
	return 1
}

func dfb_sink(value int) {} // DFB-SINK: exception-catch-sink

func run() {
	defer func() {
		if recovered := recover(); recovered != nil {
			dfb_sink(recovered.(int))
		}
	}()
	_ = dfb_source()
	panic(0) // DFB-WITNESS: exception-catch-panic
}

File: cases/taint/typescript/exception-catch-negative/ExceptionCatchNegative.ts
function dfb_source(): string { // DFB-SOURCE: exception-catch-input
  return "tainted";
}

function dfb_sink(value: string): void {} // DFB-SINK: exception-catch-sink

interface FlowError extends Error {
  value: string;
}

function run(): void {
  try {
    const flow: FlowError = new Error("flow") as FlowError;
    const ignored: string = dfb_source();
    flow.value = "clean";
    throw flow; // DFB-WITNESS: exception-catch-throw
  } catch (caught) {
    dfb_sink((caught as FlowError).value);
  }
}
